feature_engine/timeseries/forecasting/time_lag.py

```diff
@@ -0,0 +1,173 @@
+        periods: int = 1,
+        freq: str = None,
+        axis: int = 0,
+        fill_value: [int, str] = None,
+        keep_original: bool = True,
+
+    ) -> None:
+
+        if not isinstance(periods, int):
+            raise ValueError(
+                f"'num_periods' is {periods}. The variable must be "
+                f"an integer."
+            )
+
+        # add check for 'freq' param
+
+        if axis not in (0, 1):
+            raise ValueError(
+                f"'axis' is {axis}. The variable must be 0 or 1."
+            )
+
+        if not isinstance(keep_original, bool):
+            raise ValueError(
+                f"'keep_original' is {keep_original}. The variable must "
+                f"be boolean."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.periods = periods
+        self.freq = freq
+        self.axis = axis
+        self.fill_value = fill_value
+        self.keep_original = keep_original
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Creates lag
+
+        Parameters
+        ----------
+        df: pandas dataframe of shape = [n_samples, n_features]
+            The dataframe containing the variables that that will be lagged.
+
+        Returns
+        -------
+        df_new: pandas dataframe
+            The dataframe comprised of only the transformed variables or
+            the original dataframe plus the transformed variables.
+
+        """
+        # check if 'df' is a dataframe
+        _is_dataframe(df)
+
+        # check if index is a datetime object
+        # Should we check more than first index value?
+        # Could use a list comprehension to check all values, but that will be expensive.
+        if not isinstance(df.index[0], datetime.datetime):
+            raise ValueError(
+                "Dataframe's index is not a datetime object. Transformer requires"
+                "the index to be a datetime object."
+            )
+
+        # check variables
+        self.variables_ = _find_all_variables(df, self.variables)
+
+        tmp = df[self.variables_].shift(
+            periods=self.periods,
+            freq=self.freq,
+            axis=self.axis,
+            fill_value=self.fill_value
+        )
+
+        tmp.columns = self.rename_variables()
+
+        if self.keep_original:
+            df = df[self.variables_].merge(
+                tmp, left_index=True, right_index=True, how="left"
+            )
+        else:
+            df = tmp.copy()
+
+        return df
+
+    def rename_variables(self):
+        """
+        Renames variables by adding the time-lag interval.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        variables_lag: list
+            Names of the variables with the time-lag interval that is
+            used in the transformation.
+
+        """
+
+        if self.freq is None:
+            lag_str = f"_lag_{self.periods}pds"
```

Comment:
not sure pds is clear. Maybe just the period?

Also, the period is an integer, does this convert it to str?

Reply:
I wrote `pds` because I thought it would help the user delineate between time-based lags, e.g. minutes, hours, etc. These lags will be `30m`, `2h`, `1d`, etc. We could write `periods`.

And, yes, the f-string will convert `self.periods` to a string.

Reply:
I would not write anything. Otherwise variable names may become too long.